blog/views.py

```diff
@@ -0,0 +1,35 @@
+from django.contrib import messages
+from django.contrib.auth.decorators import login_required
+from django.core.paginator import Paginator
+from django.shortcuts import redirect
+from django.views import generic
+
+from blog.models import Post, Commentary
+
+
+class PostDetailView(generic.DetailView):
+    model = Post
+
+
+class PostListView(generic.ListView):
+    model = Post
+    template_name = "blog/index.html"
+    paginate_by = 5
+    context_object_name = "post_list"
+
+
+@login_required
+def add_comment(request, pk):
+    if request.method == "POST":
+        comment_content = request.POST.get("comment")
+        if comment_content:
+            post = Post.objects.get(id=pk)
```

Comment:
This line could raise a `Post.DoesNotExist` exception if there is no post with the provided `pk`. It's a good practice to handle this exception and return a 404 response. You can use Django's `get_object_or_404` function for this.